core/src/apps/nem2/metadata/layout.py

```diff
@@ -15,26 +16,36 @@
 from apps.common.layout import split_address
 
 async def ask_namespace_metadata(
-    ctx, common: NEM2TransactionCommon, namespace_metadata: NEM2NamespaceMetadataTransaction
+    ctx,
+    common: NEM2TransactionCommon,
```

Comment:
In all the layout/serialization files, we should add type hints for the common argument like: `NEM2TransactionCommon | NEM2EmbeddedTransactionCommon` cause it could be either type.